src/worldfactory.cpp

```diff
@@ -1454,12 +1454,17 @@ bool worldfactory::load_world_options(WORLDPTR &world)
 
     using namespace std::placeholders;
     const auto path = world->world_path + "/" + FILENAMES["worldoptions"];
-    if( read_from_file_optional( path, std::bind( &WORLD::load_options, world, _1 ) ) ) {
+
+    if( read_from_file_optional_jsonin( path, [world]( JsonIn &js ) { world->load_options( js ); } ) ) {
         return true;
     }
 
     const auto legacy_path = world->world_path + "/" + FILENAMES["legacy_worldoptions"];
-    if( read_from_file_optional( legacy_path, std::bind( &WORLD::load_legacy_options, world, _1 ) ) ) {
+    auto legacy_options_reader = [world]( std::istream &is ) {
+        world->load_legacy_options( is );
+    };
+
+    if( read_from_file_optional_istream( legacy_path, legacy_options_reader ) ) {
```

Comment:
That method call is part of a conditional - I don't think nesting the lambda within that conditional will enhance readability. If you prefer it that way, I will change it.